mailbox/api/src/test/java/org/apache/james/mailbox/events/GroupContract.java

```diff
@@ -0,0 +1,156 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.events;
+
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.after;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+import org.apache.james.core.User;
+import org.apache.james.mailbox.MailboxListener;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.TestId;
+import org.junit.jupiter.api.Test;
+
+import com.google.common.collect.ImmutableSortedMap;
+
+public interface GroupContract extends EventBusProvider {
```

Comment:
my proposal for this separation:
```java
interface EventBusFixutre {	
}

interface EventBusContract {
	
	interface GroupContract {
		@Test
		default void testGroup1() {
		}
	}

	@Test
	default void test1() {
	}
}

class RabbitMQEventBus implements EventBusContract.GroupContract {
}
```

With me, some sub contracts are not really big, so we could put them as a nested class. Second is, I could keep my reference as `EventBusContract.GroupContract`, it shows me GroupContract is a part of EventBusContract. Idk what other's opinions, but I find it's good.

Reply:
we need the abstract method `eventBus()`, so that I will keep my separation with inheritance